src/Servers/Kestrel/Core/src/Internal/Http/HttpHeaders.Generated.cs

```diff
@@ -3462,780 +3593,410 @@ protected override bool CopyToFast(KeyValuePair<string, StringValues>[] array, i
+                        return false;
+                    }
+                    array[arrayIndex] = new KeyValuePair<string, StringValues>("DNT", _headers._DNT);
+                    ++arrayIndex;
+                }
+                if ((_bits & 0x20000000000L) != 0)
+                {
+                    if (arrayIndex == array.Length)
+                    {
+                        return false;
+                    }
+                    array[arrayIndex] = new KeyValuePair<string, StringValues>("Upgrade-Insecure-Requests", _headers._UpgradeInsecureRequests);
+                    ++arrayIndex;
+                }
+                if ((_bits & 0x40000000000L) != 0)
+                {
+                    if (arrayIndex == array.Length)
+                    {
+                        return false;
+                    }
+                    array[arrayIndex] = new KeyValuePair<string, StringValues>("Origin", _headers._Origin);
+                    ++arrayIndex;
+                }
+                if ((_bits & 0x80000000000L) != 0)
+                {
+                    if (arrayIndex == array.Length)
+                    {
+                        return false;
+                    }
+                    array[arrayIndex] = new KeyValuePair<string, StringValues>("Access-Control-Request-Method", _headers._AccessControlRequestMethod);
+                    ++arrayIndex;
+                }
+                if ((_bits & 0x100000000000L) != 0)
+                {
+                    if (arrayIndex == array.Length)
+                    {
+                        return false;
+                    }
+                    array[arrayIndex] = new KeyValuePair<string, StringValues>("Access-Control-Request-Headers", _headers._AccessControlRequestHeaders);
+                    ++arrayIndex;
+                }
+                if (_contentLength.HasValue)
+                {
+                    if (arrayIndex == array.Length)
+                    {
+                        return false;
+                    }
+                    array[arrayIndex] = new KeyValuePair<string, StringValues>("Content-Length", HeaderUtilities.FormatNonNegativeInt64(_contentLength.Value));
+                    ++arrayIndex;
+                }
+            ((ICollection<KeyValuePair<string, StringValues>>)MaybeUnknown)?.CopyTo(array, arrayIndex);
+
+            return true;
+        }
+        
+        
+        [MethodImpl(MethodImplOptions.AggressiveOptimization)]
+        public unsafe void Append(Span<byte> name, Span<byte> value)
+        {
+            ref byte nameStart = ref MemoryMarshal.GetReference(name);
+            ref StringValues values = ref Unsafe.AsRef<StringValues>(null);
+            var flag = 0L;
+
+            // Does the name matched any "known" headers
+            switch (name.Length)
+            {
+                case 2:
+                    if (((Unsafe.Add(ref Unsafe.As<byte, ushort>(ref nameStart), 0) & 0xdfdfu) == 0x4554u))
```

Comment:
Not too worried about Endianness [as Kestrel will throw on startup](https://github.com/aspnet/AspNetCore/blob/ff313901de48b1318cf6951181f9a32ee82d8753/src/Servers/Kestrel/Core/src/KestrelServer.cs#L125-L128) if its BigEndian
```csharp
if (!BitConverter.IsLittleEndian)
{
    throw new PlatformNotSupportedException(CoreStrings.BigEndianNotSupported);
}
```
Can't really use `BinaryPrimitives` as its all span based and the amount of slicing that would occur in this method and the tmps it would cause the Jit to choke https://github.com/aspnet/AspNetCore/pull/7724#issuecomment-465375561, without making it a really weird arrangement, extra overheads aside.

Will change to ReadUnaligned though.

Reply:
Changed to ReadUnaligned; main change in `KnownHeaders.cs` in this commit https://github.com/aspnet/AspNetCore/pull/8374/commits/805505185c5b38db4659d284173b3560ea0f097d which then generates the `HttpHeaders.Generated.cs` file in same commit.